vision_datasets/visual_question_answering/vqa_as_kvp_dataset.py

```diff
@@ -0,0 +1,99 @@
+        """
+
+        if vqa_dataset is None or vqa_dataset.dataset_info.type is not DatasetTypes.VISUAL_QUESTION_ANSWERING:
+            raise ValueError("Input dataset must be a Visual Question Answering dataset.")
+
+        # Generate schema and update dataset info
+        vqa_dataset = deepcopy(vqa_dataset)
+
+        dataset_info_dict = vqa_dataset.dataset_info.__dict__
+        dataset_info_dict["type"] = DatasetTypes.KEY_VALUE_PAIR.name.lower()
+        self.img_id_to_pos = {x.id: i for i, x in enumerate(vqa_dataset.dataset_manifest.images)}
+
+        schema = self.construct_schema()
+        # Update dataset_info with schema
+        dataset_info = KeyValuePairDatasetInfo({**dataset_info_dict, "schema": schema})
+
+        # Construct KeyValuePairDatasetManifest
+        annotations = []
+        id = 1
+        for _, img in enumerate(vqa_dataset.dataset_manifest.images, 1):
+            label_data = [label.label_data for label in img.labels]
+
+            for label in label_data:
+                kvp_label_data = self.construct_kvp_label_data(label)
+                img_ids = [self.img_id_to_pos[img.id]]  # 0-based index
+                kvp_annotation = KeyValuePairLabelManifest(id, img_ids, label_data=kvp_label_data)
+                id += 1
+
+                # KVPDatasetManifest expects img.labels to be empty. Labels are instead stored in KVP annotation
+                img.labels = []
+                annotations.append(kvp_annotation)
+
+        dataset_manifest = KeyValuePairDatasetManifest(vqa_dataset.dataset_manifest.images, annotations, schema, additional_info=vqa_dataset.dataset_manifest.additional_info)
+        super().__init__(dataset_info, dataset_manifest, dataset_resources=vqa_dataset.dataset_resources)
+
+    def construct_schema(self) -> Dict[str, Any]:
+        return self.SCHEMA_BASE
+
+    def construct_kvp_label_data(self, label: Dict[str, str]) -> Dict[str, Dict[str, str]]:
+        """
+        Convert the classification dataset label_name to the desired format for KVP annotation as defined by the SCHEMA_BASE.
+        E.g. {"fields":
+                    {"answer": {"value": "yes"}},
+                    {"rationale": {"value": "I see a dog."}}
+             "text": {"question": "Is there a dog in the image?"}
+            }
+        """
+
+        if self.QUESTION_KEY not in label:
+            raise KeyError(f"Question key '{self.QUESTION_KEY}' not found in label.")
+        if self.ANSWER_KEY not in label:
+            raise KeyError(f"Answer key '{self.ANSWER_KEY}' not found in label.")
+
+        return {
+            KeyValuePairLabelManifest.LABEL_KEY: {
+                self.ANSWER_KEY: {KeyValuePairLabelManifest.LABEL_VALUE_KEY: label[self.ANSWER_KEY]},
+                self.RATIONALE_KEY: {KeyValuePairLabelManifest.LABEL_VALUE_KEY: label[self.RATIONALE_KEY] if self.RATIONALE_KEY in label else ""}
```

Comment:
is the rationale availability consistent for all samples in a dataset? If so, we might add/delete the field in the schema level. If we have it in schema but some samples don't have rationale ground truth, the evaluator will consider any rational response as wrong

Reply:
I updated the construct_kv_label_data logic to only include "rationale" if it is present in the annotation sample. 
OR
Do you think not adding "rationale" at all makes more sense? I am not sure if we want to evaluate on "rationale" key?

Reply:
yes, we can exclude it or add a boolean flag `rationale` and set to False by default.

Reply:
removed "rationale" alltogether for now. We can add back when needed.